lib/puppet/settings.rb

```diff
@@ -523,6 +523,26 @@ def preferred_run_mode=(mode)
     mode
   end
 
+  def preferred_environment_mode
+    @preferred_environment_mode || :local
+  end
+
+  # PRIVATE!  This only exists because we need a hook to validate the run mode when it's being set, and
+  #  it should never, ever, ever, ever be called from outside of this file.
+  # This method is also called when --run_mode MODE is used on the command line to set the default
+  #
+  # @param mode [String|Symbol] the name of the mode to have in effect
+  # @api private
+  def preferred_environment_mode=(mode)
+    mode = mode.to_s.downcase.intern
+    raise ValidationError, "Invalid run mode '#{mode}'" unless [:local, :remote, :not_required].include?(mode)
```

Comment:
I'd move this check to application.rb when hard-coding default environment mode there.